### DISCUSSION TITLE
Problem on MacOS 10.13.6 installing Download Helper for latest version of Downloadhelper for Firefox

### DISCUSSION THREAD

**Discussion by BaronTostoff** [aclap-dev/video-downloadhelper#1443] [Help and Q&A]
### Your question

I am trying to update downloadhelper on iMac. The Mac is running OS 10.13.6, Firefox version 115.17.0esr.
I have done this for a friend on his computer with no problems, very similar configuration.

After downloading the .PKG I get an error: 
![Screen Shot 2024-11-08 at 17 36 28](https://github.com/user-attachments/assets/cfaa35a3-326a-4835-9def-11416c9f92e9)

The same is true for downloading the .DMG file.

The --info error: "dyld: Symbol not found: ____chkstk_darwin
  Referenced from: /Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp (which was built for Mac OS X 10.15)
  Expected in: /usr/lib/libSystem.B.dylib"

Indicates that this file build is for Mac OS X 10.15, which if true would be the wrong version.

I'd appreciate any updates or other info to get this problem clarified/fixed.

Many thanks in advance!

### Addon & CoApp version and details

version: 9.1.0.44
target: mozilla
channel: stable
lang: en-GB
coapp: {"found":false,"error":"An unexpected error occurred"}
license: {"nocoapp":true}
platform: x86-64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.13; rv:109.0) Gecko/20100101 Firefox/115.0
{
    "valid_license_message_has_been_displayed": true,
    "download_directory": "/Users/maxwood/dwhelper",
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "open_count_store": 6
}

Other information requested:
CoApp

The browser you're using (be as specific as possible): Firefox 115.17.0esr (64 Bit)
OS (Linux/Windows/Mac) & OS version: MacOS High Sierra Version 10.13.6 (17G14033)
OS 64bits or 32bits: 64Bits
Link to the coapp you installed: https://github.com/aclap-dev/vdhcoapp/releases/latest/download/vdhcoapp-mac13-x86_64-installer.pkg

On Mac, the output of /Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp --info 
Applications maxwood$ /Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp --info
dyld: lazy symbol binding failed: Symbol not found: ____chkstk_darwin
  Referenced from: /Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp (which was built for Mac OS X 10.15)
  Expected in: /usr/lib/libSystem.B.dylib

dyld: Symbol not found: ____chkstk_darwin
  Referenced from: /Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp (which was built for Mac OS X 10.15)
  Expected in: /usr/lib/libSystem.B.dylib

I have deleted the /Applications/net.downloadhelper.coapp.app directory tree and all files in it, also I have deleted every instance of vdhcoapp-mac13-x86_64-installer that I have downloaded
rm -R net.downloadhelper.coapp.app
rm: net.downloadhelper.coapp.app: No such file or directory

/Applications  ls net.downloadhelper.coapp.app
ls: net.downloadhelper.coapp.app: No such file or directory

Anti-Virus (malwarebytes) switched off for testing.

**Comment by Mattmjs** [comment]
Your system can only run an old version of the companion app. That includes dropping back to 
an older extension and disabling auto updates.
Here is the link, you need 8.0.0.6 + 1.6.3 

https://github.com/aclap-dev/video-downloadhelper/wiki/Older-versions

**Comment by BaronTostoff** [comment]
Hi,

Many thanks for this information which has worked to fix my problem.

Just a note, I initially downloaded the .PKG version of the coapp and it didn’t work (same error about needing to install the coapp) however the .DMG version worked fine, indeed at the point of “moving” the coapp to the Applications folder I was prompted asking if I really wanted to replace the NEWER file with this one (from the DMG)

Sorry if that’s not clear, it seems that the two installation options contain different versions of the coapp.

Again, thanks,

Case closed ;-)
